app/src/test/kotlin/com/intive/picover/common/viewmodel/state/ViewModelStateTest.kt

```diff
@@ -0,0 +1,54 @@
+package com.intive.picover.common.viewmodel.state
+
+import io.kotest.assertions.assertSoftly
+import io.kotest.core.spec.style.ShouldSpec
+import io.kotest.inspectors.forAll
+import io.kotest.matchers.shouldBe
+import io.mockk.mockk
+
+internal class ViewModelStateTest : ShouldSpec(
+	{
+		val data: Any = mockk()
+
+		should("SHOULD isLoading AND isError AND isLoaded AND data return proper values based on view model state") {
```

Comment:
minor: I don't see to point any test with `proper`  in a name, we can actually apply it to every operation because everything should be done in a "proper" way :)

Reply:
i can agree on that :D fixed